src/RequestContext.js

```diff
@@ -40,12 +40,23 @@ module.exports = class RequestContext {
     this._wskActivationId = utils.randomChars(32, true);
     this._requestId = utils.randomChars(32);
     this._cdnRequestId = utils.uuid();
-    this._strain = cfg.selectStrain({ ...req, path: this._path });
-    if (this._strain.urls.length > 0) {
-      this._mount = parse(this._strain.urls[0]).pathname.replace(/\/+$/, '');
+
+    if (cfg.matchStrain) {
```

Comment:
why backward comp. check?

Reply:
There are tests that mock up selectStrain() (`utils_test.js` and `template_resolver_tests.js`), and I didn't want to touch those.

Reply:
> and I didn't want to touch those.

why not?

Reply:
Because if I add `matchStrain` to them as well, there will be no more coverage for `selectStrain` at all, so we could just as well get rid of it.

As I noticed lately, I could alternatively change `selectStrain`'s signature to include an extra `opts={}` parameter, which would behave like `matchStrain` if `opts` includes `returnMatch`, what do you think?

Reply:
you could also just change the behaviour and introduce a BREAKING CHANGE.
the only place we use HelixProject is in helix-cli, I tihnk, and it doesn't need `selectStrain`:

https://github.com/adobe/helix-cli/search?q=selectStrain&unscoped_q=selectStrain

Reply:
Changed, please review again :)